### DISCUSSION TITLE
Write multiple files with tool script

### DISCUSSION THREAD

**Discussion by ataube** [cue-lang/cue#1248] [Q&A] [answered]
Hi there, 
I would like to create a tool script which iterates on a list of data and writes each entry as a dedicated json file to disk. 
Something like this

```cue
command: dump: {
  for x in events {
    write: file.Create & {
      filename: "\(x.eventId).json"
      contents: json.Marshal(x)
    }
  }
}
```

Do you have an idea why it is not working? I'm getting the following error
```
command.dump.write.filename: conflicting values "4a714c16-6342-4a38-a0f0-d9c92648ee5d.json" and "afb37fa1-499f-4099-a10d-b59a061ff63b.json":
    ./events_tool.cue:16:3
    ./events_tool.cue:18:17
```

**Comment by ataube** [comment, marked as answer]
I just found this helpful issue and got it to work: https://github.com/cuelang/cue/issues/67
Sorry for the noise

**Reply by verdverm** [reply]
As a quick reference, the main idea is that when you are generating a `label: { ... }` struct within a for loop (struct comprehension), the key needs to be unique, otherwise the structs created at each iteration unify together.

```
command: foo: {
  for key, val in listOrStruct {
    "write-\(key): file.Create & {    // <== note the interpolation here, this label needs to be unique
      filename: "\(val.name).json"
      contents: json.Marshal(val)
    }
  }
}
```
You could also omit the `key` above and use a field from `val` for constructing a unique key
